src/main/java/io/anserini/kg/freebase/EntityLinking.java

```diff
@@ -0,0 +1,360 @@
+  }
+
+  /**
+   * Prints query results to the standard output stream.
+   *
+   * @param queryName the entity name to search
+   * @throws Exception on error
+   */
+  public List<RankedEntity> search(String queryName) throws Exception {
+    // Initialize index searcher
+    IndexSearcher searcher = new IndexSearcher(reader);
+
+    Similarity similarity = new ClassicSimilarity();
+    searcher.setSimilarity(similarity);
+
+    MultiFieldQueryParser queryParser = new MultiFieldQueryParser(
+            new String[]{ IndexTopics.FIELD_NAME,
+                    IndexTopics.FIELD_LABEL },
+            new SimpleAnalyzer());
+    queryParser.setDefaultOperator(QueryParser.Operator.AND);
+    Query query = queryParser.parse(queryName);
+
+    TopDocs rs = searcher.search(query, numHits);
+    ScoredDocuments docs = ScoredDocuments.fromTopDocs(rs, searcher);
+    List<RankedEntity> rankedEntities = new ArrayList<>(numHits);
+
+    for (int i = 0; i < docs.documents.length; i++) {
+      float score = docs.scores[i];
+      String mid = docs.documents[i].getField(IndexTopics.FIELD_TOPIC_MID).stringValue();
+      String shortMid = getShortMid(mid);
+      String name = docs.documents[i].getField(IndexTopics.FIELD_NAME).stringValue();
+      String label = docs.documents[i].getField(IndexTopics.FIELD_LABEL).stringValue();
+      rankedEntities.add( new RankedEntity(shortMid, score, name, label) );
+    }
+
+    return rankedEntities;
+  }
+
+  // Removes '<', '>' if they exist, lower case
+  private static String cleanUri(String uri) {
+    if (uri.charAt(0) == '<')
+      return uri.substring(1, uri.length() - 1).toLowerCase();
+    else
+      return uri;
+  }
+
+  // get 1-gram, 2-gram, 3-gram, 4-gram combinations from the text
+  private static List<String> getSearchQueries(String text) {
+    List<String> ngrams = new ArrayList<String>();
+    if (text.endsWith(".") || text.endsWith("?")) {
+      text = text.substring(0, text.length()-1);
+    }
+    String[] tokens = text.split("\\s+");
+
+    // add unigram
+    for (int i=0; i < tokens.length; i++) {
+      ngrams.add(tokens[i]);
+    }
+
+    // add bigram
+    for (int i=1; i < tokens.length; i++) {
+      ngrams.add(tokens[i-1] + " " + tokens[i]);
+    }
+
+    // add 3-gram
+    for (int i=2; i < tokens.length; i++) {
+      ngrams.add(tokens[i-2] + " " + tokens[i-1] + " " + tokens[i]);
+    }
+
+    // add 4-gram
+    for (int i=3; i < tokens.length; i++) {
+      ngrams.add(tokens[i-3] + " " + tokens[i-2] + " " + tokens[i-1] + " " + tokens[i]);
+    }
+
+    return ngrams;
+  }
+
+  // get short Mid from the long URL - needed for comparison
+  private static String getShortMid(String text) {
+    if(text.startsWith("www.freebase.com/")) {
+      return text.substring(17).replaceAll("/", ".");
+    }
+    else if(text.startsWith("http://rdf.freebase.com/ns/")) {
+      return text.substring(27);
+    }
+    else if(text.startsWith("fb:")) {
+      return text.substring(3);
```

Comment:
After replacing with constant string field, use `text.substring(const_str.length())`, you might want to combine with previous condition too.